test/caption-stream.test.js

```diff
@@ -1460,6 +1465,112 @@ QUnit.test('the roll-up count can be changed on-the-fly', function() {
   QUnit.equal(captions.length, 0, 'cleared the caption');
 });
 
+QUnit.test('switching to roll-up from pop-on wipes memories and flushes captions', function() {
+  var captions = [];
+  cea608Stream.on('data', function(caption) {
+    captions.push(caption);
+  });
+
+  [
+    { pts: 0 * 1000, ccData: 0x1420, type: 0 },
+    { pts: 0 * 1000, ccData: characters('hi'), type: 0 },
+    // flip memories
+    { pts: 1 * 1000, ccData: 0x142f, type: 0 },
+    { pts: 1 * 1000, ccData: 0x1420, type: 0 },
+    { pts: 2 * 1000, ccData: characters('oh'), type: 0 },
+    { pts: 2 * 1000, ccData: 0x142f, type: 0 },
+    { pts: 3 * 1000, ccData: 0x1425, type: 0 }
+  ].forEach(cea608Stream.push, cea608Stream);
+
+  var displayed = cea608Stream.displayed_.reduce(function(acc, val) {
+    acc += val.trim();
```

Comment:
not sure if it matters for this test (or any of the others that you do this `reduce`), but I don't think the first value in the array will have `trim()` called on it.

Reply:
Ah yeah good point. Thankfully I don't think I need the `trim` here, so I just removed it.